### PR TITLE
Optional league reorg

### PR DESCRIPTION
**Pull request by cpfergus1**
Reorganized league methods to game_teams

### CODE REVIEW

**Comment by cpfergus1** [comment]
Resolved conflicts for merge to successfully join

File: fixtures/fixture_game_teams.csv [hunk 1]
```diff
@@ -1,81 +1,499 @@
...
-2015020606,10,away,LOSS,REG,Mike Babcock,1,6,24,19,4,1,56.9,6,2
-2015020606,26,home,WIN,REG,Darryl Sutter,2,10,23,10,2,0,43.1,12,2
-2017020240,23,away,LOSS,REG,Travis Green,1,5,11,27,5,0,47.5,13,5
-2017020240,24,home,WIN,REG,Randy Carlyle,2,8,10,15,6,3,52.5,11,6
-2017020624,29,away,LOSS,REG,John Tortorella,0,8,12,8,3,0,57.6,3,8
-2017020624,21,home,WIN,REG,Jared Bednar,2,8,20,8,3,1,42.4,5,5
-2015020346,22,away,TIE,SO,Todd McLellan,2,9,31,8,4,1,50.6,5,2
-2015020346,5,home,TIE,SO,Mike Johnston,2,10,36,8,4,1,49.4,10,5
-2016020205,53,home,LOSS,REG,Dave Tippett,2,6,35,9,1,0,51,9,9
+2012020001,4,home,LOSS,REG,Peter Laviolette,1,6,40,6,5,0,43.5,12,8
```
Comment (c-bartell) [inline fixtures/fixture_game_teams.csv:2]: I think these additions are good. Do all of the test files still work?

File: lib/game_team_manager.rb [hunk 2]
```diff
@@ -15,7 +15,48 @@ def generate_game_teams(location) # I need a test
...
+
+  def lowest_scoring_home_team
+    team_data.min_by{|team| goal_avg_per_team(team.team_id, 'home')}.team_name
+  end
+
+  def team_data
+    @stat_tracker.team_manager.teams
+  end
 
+=======
```
Comment (c-bartell) [inline lib/game_team_manager.rb:59]: You might have lost an end here in a merge resolution.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -104,27 +104,27 @@ def count_of_teams
   end
 
   def best_offense
-    @game_manager.best_offense
+    @game_teams_manager.best_offense
```
Comment (c-bartell) [inline lib/stat_tracker.rb:107]: I changed all of these when I was debugging and I was wrong :(. I think the consensus was to make all the references `@game_team_manager`. That was my bad.

File: lib/game_team_manager.rb [hunk 4]
```diff
@@ -15,7 +15,48 @@ def generate_game_teams(location) # I need a test
     end
     array
   end
+  
+  def goal_avg_per_team(team_id, home_away)
```
Comment (c-bartell) [inline lib/game_team_manager.rb:19]: I think you're spot on with this. It makes more sense to have these in GameTeamsManager. 👍

**Review by c-bartell — APPROVED** [review]
I think it makes sense to move these methods. As long as the tests and rspec pass, this has my vote to merge!

**Review by s-steel — APPROVED** [review]
Like it.  Looks good to move all that over to the game_team_manager.

**Pull request merged**